python/tskit/trees.py

```diff
@@ -9291,25 +9290,156 @@ def ibd_segments(
             store_pairs=store_pairs,
         )
 
-    def coalescence_time_distribution(
+    def coalescing_pairs(
         self,
-        *,
         sample_sets=None,
-        weight_func=None,
-        window_breaks=None,
-        blocks_per_window=None,
-        span_normalise=False,
+        indexes=None,
+        windows=None,
+        span_normalise=True,
     ):
-        # TODO docstring, not yet in API
+        """
+        Calculate the number of coalescing sample pairs per node, summed over
+        trees and weighted by tree span.
 
-        return stats.CoalescenceTimeDistribution(
-            self,
-            sample_sets=sample_sets,
-            weight_func=weight_func,
-            window_breaks=window_breaks,
-            blocks_per_window=blocks_per_window,
-            span_normalise=span_normalise,
-        )
+        The number of coalescing pairs may be calculated within or between the
+        non-overlapping lists of samples contained in `sample_sets`. In the
+        latter case, pairs are counted if they have exactly one member in each
+        of two sample sets. If `sample_sets` is omitted, a single group containing
+        all samples is assumed.
+
+        The argument `indexes` may be used to specify which pairs of sample
+        sets to compute the statistic between, and in what order. If omitted,
+        all possible pairs of sample sets are output in lexicographic order
```

Comment:
Hm, this is somewhat different than [in other stats](https://tskit.dev/tskit/docs/latest/stats.html#output-dimensions) - if we followed that, then if there were two sample sets and `indexes=None` that'd be the same as `indexes=(0,1)`; while if there weren't two sample sets then `indexes=None` would produce an error.

Reply:
A nice thing about making people input the indexes explicitly rather than letting them rely on `indexes=None` is then they can't get mixed up about what order the output is coming in (since the output is not labeled).

Reply:
that'd be good to match. In which case, if there's one sample set and indexes=None, then I think that should be equivalent to indexes=(0,0)?